src/nav/nav.ts

```diff
@@ -0,0 +1,287 @@
+  @Output() activeIdChange = new EventEmitter<any>();
+
+  /**
+   * If `true`, non-active nav content will be removed from DOM
+   * Otherwise it will just be hidden
+   */
+  @Input() destroyOnHide;
+
+  /**
+   * The orientation of navs.
+   *
+   * Using `vertical` will also add the `aria-orientation` attribute
+   */
+  @Input() orientation: 'horizontal' | 'vertical';
+
+  /**
+   * Role attribute generating strategy:
+   * - `false` - no role attributes will be generated
+   * - `'tablist'` - 'tablist', 'tab' and 'tabpanel' will be generated (default)
+   */
+  @Input() roles: 'tablist' | false;
+
+  @ContentChildren(NgbNavItem) items: QueryList<NgbNavItem>;
+
+  constructor(@Attribute('role') public role: string, config: NgbNavConfig, private _cd: ChangeDetectorRef) {
+    this.destroyOnHide = config.destroyOnHide;
+    this.orientation = config.orientation;
+    this.roles = config.roles;
+  }
+
+  /**
+   * The nav change event emitted right before the nav change happens
+   *
+   * See `NgbNavChangeEvent` for payload details
+   */
+  @Output() navChange = new EventEmitter<NgbNavChangeEvent>();
+
+  click(item: NgbNavItem) {
+    if (!item.disabled && !item.active) {
+      this._updateActiveId(item.id);
+    }
+  }
+
+  /**
+   * Selects the nav with the given id and shows its associated pane.
+   * Any other nav that was previously selected becomes unselected and its associated pane is hidden.
+   */
+  select(id: any) {
+    if (this.activeId !== id) {
+      this._updateActiveId(id);
+    }
+  }
+
+  ngAfterContentInit() {
+    if (!isDefined(this.activeId)) {
+      const nextId = this.items.first ? this.items.first.id : null;
+      if (nextId) {
+        this._updateActiveId(nextId);
+        this._cd.detectChanges();
+      }
+    }
+  }
+
+
+  ngOnChanges({activeId}: SimpleChanges): void {
+    if (activeId && !activeId.isFirstChange()) {
+      const {previousValue, currentValue} = activeId;
+      const defaultPrevented = this._updateActiveId(currentValue, previousValue);
+
+      // This will happen only in the case when [activeId] was changed and (navChange) was prevented at the same time.
+      // It is here to keep consistency for changing activeId with [activeId], .select() and (click).
+      if (defaultPrevented) {
+        this.activeId = previousValue;
+        Promise.resolve(true).then(() => { this.activeIdChange.emit(previousValue); });
+      }
+    }
+  }
+
+  private _updateActiveId(nextId: any, activeId = this.activeId) {
+    let defaultPrevented = false;
+
+    if (activeId !== nextId) {
+      this.navChange.emit({activeId, nextId, preventDefault: () => { defaultPrevented = true; }});
+    }
+
+    if (!defaultPrevented) {
+      this.activeId = nextId;
+      this.activeIdChange.emit(nextId);
+    }
+
+    return defaultPrevented;
+  }
+}
+
+
+/**
+ * A directive to put on the nav link.
+ */
+@Directive({
+  selector: 'a[ngbNavLink]',
+  host: {
+    '[id]': 'navItem.domId',
+    'class': 'nav-link',
```

Comment:
I think we should use `'[class.nav-link]': 'true'` for the class binding to avoid any collision and/or override

Reply:
@benouat I don't think `'class': 'xxx'` has any problems. Issues with overriding happen when you do either `[class]: ...` or `class: {{ ... }}`.

Anyway instead I:
- fixed `ngbNav` to have `'class': 'nav'` instead of `'[class.nav]': 'true'`
- updated a test to check that user can override classes for all items (nav, link, item, outlet)

So thanks :)

Reply:
@benouat Hmm, I just realized that it works both ways :) and your comment is completely valid. Fixed both cases and made a better test.

Reply:
You just saved me a Stackblitz! Thanks 👏